### PR TITLE
override styling for inline code when in night-theme (#2210)

### PR DESCRIPTION
**Pull request by ashToronto**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Documentation Update

## Description
Fixed the night-theme mode color contrast for inline code comments

## Related Tickets & Documents
#2210 

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)
Screen Shot 2019-04-12 at 12 46 46 PM

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [ ] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: app/assets/stylesheets/variables.scss [hunk 1]
```diff
@@ -21,6 +21,7 @@ $yellow: #fefa87;
 $tan: #fdf9f3;
 $off-white: #fffffe;
 $bold-blue: #0045ff;
+$night-theme-markdown-inline-code: #ffffff;
```
Comment (venarius) [inline app/assets/stylesheets/variables.scss]: There are already variables which are defined by each theme. There is a var called `--theme-container-color` which defines `#fff` for the dark theme.

File: app/assets/stylesheets/variables.scss [hunk 2]
```diff
@@ -40,6 +41,9 @@ body.night-theme {
     -webkit-filter: invert(95%);
     filter: invert(95%);
   }
+  .inner-comment code {
```
Comment (venarius) [inline app/assets/stylesheets/variables.scss]: I would rather update the `.single-comment-node .body code` selector which is already defined in `comments.scss`

**Comment by Link2Twenty** [comment]
This doesn't quite work with our current theming model. You'll need to add a line under this line that uses the css variable.

https://github.com/thepracticaldev/dev.to/blob/d39d30b8fdc8d27fca47579b77e83672f56b8da7/app/assets/stylesheets/comments.scss#L624

```css
  color: var(--theme-color);
```

This change will make code blocks in comments look like this

image

Also, it will make the change compatible with future themes.

Thanks 🙂

**Comment by omenking** [comment]
@Link2Twenty  We neeed some help understanding the `current theming model`
Can you provide us with more information and could we improve the documentation for others.


[](http://www.youtube.com/watch?v=7uiAhQ9pbrA "")

**Comment by Link2Twenty** [comment]
Hi @omenking, sorry for the lack of documentation. I had been spearheading the theming project, which went live around 4 weeks ago, but I started a new just a few days after it went live meaning I didn't have time to document things as well as I'd like, for which I am truly sorry.

@benhalpern did a blog post a while ago explaining how to theme things. https://dev.to/devteam/contributors-wanted-night-mode-themes-on-dev-i8n which I know is not where you'd expect to find documentation.

If you need anything else clarifying feel free to ask here or on the post (tag ben or me).

**Comment by rhymes** [comment]
Hi @omenking I remember from my adventures in the code there's a place where the theme variables are defined. I don't know how the overall feature works exactly but that might be a starting point.

https://github.com/thepracticaldev/dev.to/blob/573cc2eb1b8feb2b78a1990ec7d9e79144c7370d/app/views/layouts/_user_config.html.erb#L1-L29

Hope this helps a bit

**Comment by venarius** [comment]
@rhymes thats pretty much how the theming for dev.to works as as far as I get it.

**Comment by Link2Twenty** [comment]
@rhymes that's where the themes are stored at the moment but they will be moved at some point 🙂

**Comment by omenking** [comment]
@Link2Twenty thanks for the quick response. Will see if we can get some documentation on the styling organized for the project.

**Comment by maestromac** [comment]
Superseded by https://github.com/thepracticaldev/dev.to/pull/2947

**Pull request closed without merging**